engine/runtime/src/main/scala/org/enso/interpreter/runtime/scope/LocalScope.scala

```diff
@@ -162,6 +163,18 @@ class LocalScope(
         )
       assert(localFrameSlotIdxs(definition.id) == returnedFrameIdx)
     }
+    val ctx =
+      try {
+        EnsoContext.get(null)
+      } catch {
+        case _: RuntimeException => null
+      }
+    val defValue = if (ctx != null) {
+      ctx.getBuiltins().error().dataFlowErrorUninitializedValue()
+    } else {
+      0L;
```

Comment:
This is a nasty part. Some `FrameDescriptor`s are created sooner than `EnsoContext` is initialized - for example for builtins. In such case we don't have the `dataFlowErrorUninitializedValue()` to put it in and we put `0L` there. Not ideal. On the other hand, errors in builtins are under our control and shall be fixed properly, not just by putting in default value.

Moreover sometimes `EnsoContext.get` can even throw an exception... hence the `try/catch` above.

Reply:
Shouldn't we rather fix `ReadLocalVariableNode` to throw a `dataFlowErrorUnitializedValue` once it tries to read a frame slot that has `Illegal` kind? You wouldn't need to do this weird hack, because in `ReadLocalVariableNode`, you can be sure that there is an `EnsoContext`.

Reply:
Possible, but it will require a NPE check. The solution with _default value_ would be for "free".

Reply:
> This is a nasty part. Some `FrameDescriptor`s are created sooner than `EnsoContext`

Nasty code is gone in 40dfa45 - we use _context-less_ `DataflowError.UNINITIALIZED` value which is available all the time.